python/neutron-understack/neutron_understack/neutron_understack_mech.py

```diff
@@ -263,14 +263,16 @@ def __network_name(self, network_id: str):
             return "tenant"
 
     def _move_to_network(self, context):
+        interface_uuid = context.current["id"]
```

Comment:
I am afraid this won't be as simple, but I am not 100% sure on that.

The way I understand it: 
- the ml2 driver calls [`update_port_postcommit`](https://docs.openstack.org/neutron-lib/latest/reference/modules/neutron_lib.plugins.ml2.api.html#neutron_lib.plugins.ml2.api.MechanismDriver.update_port_postcommit) with a relevant context that is of [PortContext](https://docs.openstack.org/neutron-lib/latest/reference/modules/neutron_lib.plugins.ml2.api.html#neutron_lib.plugins.ml2.api.PortContext) type
- this gets passed to `_move_to_network` without modifications

Now, the problem I see here is that `id` fields corresponds to the ID of a port as given by Neutron. Unless I'm wrong, these are completely different to what Nautobot expects to identify an Interface.

These IDs are automatically assigned when Ironic [creates](https://opendev.org/openstack/ironic/src/branch/master/ironic/common/neutron.py#L359) Neutron ports for each of its [baremetal Ports](https://docs.openstack.org/ironic/latest/contributor/api/ironic.objects.port.html#ironic.objects.port.Port).

At the moment, both baremetal ports and neutron port IDs will not match as they are both randomly allocated - on baremetal side during the node inspection, and on the neutron side when the code linked above is executed.

I think the easiest way around that situation may be to match using the MAC address. Alternatively, we will have to pre-create the neutron ports with specific IDs which may be tricky.

Reply:
One clarification on this - even though Ironic initially [copies the baremetal port attributes](https://opendev.org/openstack/ironic/src/commit/3f22f8c5f00e6a1af9f221df9b22698a02baa245/ironic/common/neutron.py#L305) presumably including the baremetal port id, it won't be preserved because Python Openstack SDK (which is used to interface with Neutron), [does not pass on](https://docs.openstack.org/openstacksdk/latest/user/connection.html#openstack.connection.Connection.create_port) that attribute.

Even if it did, [Neutron API](https://docs.openstack.org/api-ref/network/v2/#create-port) wouldn't accept it

```
root@openstack-admin-client:/tmp# cat create_port_with_id.json
{
    "port": {
        "id": "bc1ab041-13af-464c-81c6-87db1e15aaaa",
        "admin_state_up": true,
        "name": "private-port",
        "network_id": "a0c267bf-33af-48b9-b540-daa6d41bff3f",
        "port_security_enabled": false,
        "allowed_address_pairs": []
    }
}

root@openstack-admin-client:/tmp# curl -X POST -H "Content-Type: application/json" -H "X-Auth-Token: $TOKEN" http://neutron-server.openstack.svc.cluster.local:9696/v2.0/ports -d @create_port_with_id.json
{"NeutronError": {"type": "HTTPBadRequest", "message": "Attribute 'id' not allowed in POST", "detail": ""}}root@openstack-admin-client:/tmp#
root@openstack-admin-client:/tmp#
```

Reply:
I thought it was too good to be true when someone said "that is the ID you need".  I reworked this in 35d1295 to find the interface by MAC address, we can revisit this in the future if we ever sync up the UUID values.

Reply:
@skrobul We can go directly to python-neutronclient instead of using the OpenStack SDK. I've noticed the OpenStack SDK is a bit too high level in a number of places. It aims to be a higher level simple interface and I guess that makes sense.

Reply:
My comments doesn't matter because like you said, neutron rejects that.